ethstore/src/accounts_dir/disk.rs

```diff
@@ -153,8 +153,42 @@ impl<T> DiskDirectory<T> where T: KeyFileManager {
 		)
 	}
 
-	/// insert account with given file name
-	pub fn insert_with_filename(&self, account: SafeAccount, filename: String) -> Result<SafeAccount, Error> {
+	/// insert account with given file name. if the filename is a duplicate of any stored account, a number is appended
+	/// to the filename.
+	pub fn insert_with_filename(&self, account: SafeAccount, mut filename: String) -> Result<SafeAccount, Error> {
+		// check for duplicate filenames and append/increment counter
+		let dups: Vec<_> = self.files()?.into_iter().filter_map(|f| {
```

Comment:
Also - maybe just check if file exists (`fs::metadata(_)is_ok()`) and then just add a random suffix, instead of reading all the files list (`self.files()`) and trying to maintain an order (do not know if an order is important here)? This should increase performance of insert/update - iirc there are cases when there are a lot (>>10000) key files in single dir.
Do not have a strong opinion on this, though